packages/admin-panel-server/src/viz-builder/dashboardVisualisation/reportConfigValidator.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
+ */
+
+import { yup, yupUtils } from '@tupaia/utils';
+
+// TODO avoid duplication with report-server
```

Comment:
I wanted to get your opinion on that... Obviously we don't want to duplicate this code, but I don't if we want to add `@tupaia/server-server` as a dependency in `@tupaia/admin-panel-server` for the sole purpose of using its validation. I'm thinking that if we eventually deploye those as microservices, we may want them to be entirely separated.

Any ideas?

Reply:
The main motivation for validation here is that it is used while importing/exporting vizes here, were report-server is not involved.

And that brings me to a suggestion: how about we add a validation endpoint in report-server?

Reply:
Coolz, I created https://linear.app/bes/issue/PHX-108/make-report-validation-dry